src/api/localStorage.js

```diff
@@ -0,0 +1,111 @@
+import Enumerable from '../../node_modules/linq';
+import faker from '../../node_modules/faker';
+export class LocalStorage {
+  constructor(_id) {
+    this._id = _id;
+  }
+
+  createUser() {
+    let existingUsers = JSON.parse(localStorage.getItem('userAccounts'));
+    if (existingUsers === null) existingUsers = [];
+    this._id = Math.floor(10000 + Math.random() * 9000);
+    let userName = faker.fake('{{name.firstName}} {{name.lastName}}');
+
+    let userEntry = {
+      '_id': this._id,
+      'name': userName
+    };
+    
+    existingUsers.push(userEntry); 
+    localStorage.setItem('userAccounts', JSON.stringify(existingUsers));
+    console.log('createUser() -> User Added: ', userEntry);
+  }
+  
+  getUser(id) {
+    let users = JSON.parse(localStorage.getItem('userAccounts'));
+
+    let user = Enumerable
+      .from(users)
+      .where((u) => u._id == this._id)
+      .firstOrDefault();
+
+    console.log('getUser() -> User Found: ', user);
+    return user;
+  }
+
+  getAllUsers() {
+    console.log('getAllUsers() -> Users Found: ',
+     JSON.parse(localStorage.getItem('userAccounts')));
+    return JSON.parse(localStorage.getItem('userAccounts'));
+  }
+
+  updateUser(id, update) {
+    let existingUsers = JSON.parse(localStorage.getItem('userAccounts'));
+    for (let i = 0; i < existingUsers.length; i++) {
+      if (existingUsers[i]._id === this._id) {
+        const updatedUser = Object.assign(existingUsers[i], update);
+        console.log('updateUser() -> Updated User: ', updatedUser);
+      }
+    }
+  }
+
+  replaceUser(id, update) {
+    let users = JSON.parse(localStorage.getItem('userAccounts'));
+    for (let i = 0; i < users.length; i++) {
```

Comment:
should use the simplified syntax
```suggestion
    users.forEach({
```

Reply:
Better yet
```suggestion
    const index = users.indexOf(users.find(user => user._id === id));
    Object.assign(users[index], update);
```